### PR TITLE
Revamp `transforming message formats` tutorial

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Comment by mindula** [comment]
(L17) let's add a full stop.

**Comment by mindula** [comment]
(L255) should be reservation not payload

**Comment by mindula** [comment]
(L255) The parameter name should be reservation not payload.

**Comment by mindula** [comment]
(L262-L267) We decided to remove log statements except for error logs.

**Comment by mindula** [comment]
(L60) Shall we change the appointment number to 1?

**Comment by mindula** [comment]
(L351) Should be [reservation.hospitalId]

File: swan-lake/integration-tutorials/transforming-message-formats.md [hunk 1]
```diff
@@ -101,48 +100,125 @@ Follow the instructions given in this section to develop the service.
...
-    > ```ballerina
-    > final http:Client hospitalServiceEP = new (hospitalServiceUrl);
-    >```
+    > **Note:**
+    > While it is possible to work with the JSON payload directly, using record types offers several advantages including enhanced type safety, data validation, and better tooling experience (e.g., completion).
+
+    > **Note:**
+    > When the fields of the JSON objects are expected to be exactly those specified in the sample payload, the generated records can be updated to be [closed records](https://ballerina.io/learn/by-example/controlling-openness/), which would indicate that no other fields are allowed or expected.
+
+4. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and sends an email to the client.
```
Comment (mindula) [inline swan-lake/integration-tutorials/transforming-message-formats.md:193]: Should not add the `sends an email..` part

File: swan-lake/integration-tutorials/transforming-message-formats.md [hunk 2]
```diff
@@ -162,62 +238,28 @@ Follow the instructions given in this section to develop the service.
             appointment_date: reservation.appointmentDate
         };
     ```
+6. Define a [`http:Client`](https://ballerina.io/learn/by-example/#http-client) object to send requests to the backend service.
```
Comment (mindula) [inline swan-lake/integration-tutorials/transforming-message-formats.md:241]:
```suggestion
6. Define an [`http:Client`](https://ballerina.io/learn/by-example/#http-client) object to send requests to the backend service.
```

**Review by mindula — APPROVED** [review]

File: swan-lake/integration-tutorials/transforming-message-formats.md [hunk 3]
```diff
@@ -101,48 +100,125 @@ Follow the instructions given in this section to develop the service.
...
     ```
+2. Remove the generated content in the `main.bal` file and open the diagram view in VS Code.
 
-2. Introduce the source code in files with the `.bal` extension (e.g., the `main.bal` file).
+    ![Open diagram view](/learn/images/integration-tutorials/transforming-message-formats/open_diagram_view.gif)
 
-    Import the
-    - `ballerina/http` module to develop the REST API and define the client that can be used to send requests to the backend services
-    - `ballerina/log` module to log debug, error, or info level information for each client request
+3. Generate record types corresponding to the payloads from the hospital service and input/output of the transforming function by providing samples of the expected JSON payloads.
```
Comment (MaryamZi) [inline swan-lake/integration-tutorials/transforming-message-formats.md:107]: We haven't spoken about a "transforming function" before we get here, so may not make sense to the user.

I would have 

1. if there is a request payload, show generating the record for that. if not, show generating a record for the payload from the backend.
2. if there are other request or response payloads, say "Similarly, generate records ...". Don't have to give the JSON payloads here.
3. then say "The generated record types will be as follows." for all.

So basically, show one JSON -> record generation for whatever we encounter first. And then say generate similarly for others and show the records.

File: swan-lake/integration-tutorials/transforming-message-formats.md [hunk 4]
```diff
@@ -101,48 +100,125 @@ Follow the instructions given in this section to develop the service.
...
-    > ```ballerina
-    > final http:Client hospitalServiceEP = new (hospitalServiceUrl);
-    >```
+    > **Note:**
+    > While it is possible to work with the JSON payload directly, using record types offers several advantages including enhanced type safety, data validation, and better tooling experience (e.g., completion).
+
+    > **Note:**
+    > When the fields of the JSON objects are expected to be exactly those specified in the sample payload, the generated records can be updated to be [closed records](https://ballerina.io/learn/by-example/controlling-openness/), which would indicate that no other fields are allowed or expected.
+
+4. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request.
```
Comment (MaryamZi) [inline swan-lake/integration-tutorials/transforming-message-formats.md:193]: Better to include something about transforming here.

File: swan-lake/integration-tutorials/transforming-message-formats.md [hunk 5]
```diff
@@ -233,40 +275,11 @@ Follow the instructions given in this section to develop the service.
...
-
-        ```ballerina
-        log:printError("Reservation request failed", resp);
-        if resp is http:ClientRequestError {
-            return <http:NotFound> {body: "Unknown hospital, doctor or category"};
-        }
-
-        return <http:InternalServerError> {body: resp.message()};
-        ```
+    - If the response is not a `ReservationResponse` record, log the information at `ERROR` level. Return a "NotFound" response if the response from the hospital service is an `http:ClientRequestError` response or an "InternalServerError" response otherwise.
```
Comment (MaryamZi) [inline swan-lake/integration-tutorials/transforming-message-formats.md:282]: Fix "NotFound" -> `http:NotFound`.

Same for "InternalServerError".

File: swan-lake/integration-tutorials/transforming-message-formats.md [hunk 6]
```diff
@@ -101,48 +100,125 @@ Follow the instructions given in this section to develop the service.
...
-    >
-    > ```ballerina
-    > final http:Client hospitalServiceEP = new (hospitalServiceUrl);
-    >```
+    > **Note:**
+    > While it is possible to work with the JSON payload directly, using record types offers several advantages including enhanced type safety, data validation, and better tooling experience (e.g., completion).
+
+    > **Note:**
+    > When the fields of the JSON objects are expected to be exactly those specified in the sample payload, the generated records can be updated to be [closed records](https://ballerina.io/learn/by-example/controlling-openness/), which would indicate that no other fields are allowed or expected.
+
```
Comment (MaryamZi) [inline swan-lake/integration-tutorials/transforming-message-formats.md:192]: Add a note about making ssn an array of fixed-length (and maybe link to a BBE).
